Put the auto-generation script under version control

sbesson opened the pull request:
This PR puts the script that is used by the auto-generation jobs (http://ci.openmicroscopy.org/job/OMERO-5.1-merge-docs-autogen/ and http://ci.openmicroscopy.org/job/OMERO-5.1-latest-docs-autogen/) under version control to simplify its improvements by Pull Requests. 

An extra feature of this PR is to do the substitution of the CI WORKSPACE by `/home/omero` across the board (i.e. including in the configuration properties page).

joshmoore commented:
Big :thumbsup: for the idea. Will the merge job be modified to test this in place? (Or has it already been?)

sbesson commented:
At the moment, both merge and latest jobs are using the same version of the script. Another thought with this job is that the `merge` job would also use the merge version of this script allowing to review the output of any PR opened against it. This would have the implication that the `merge` autogen branch may not be mergeable as opposed to the current state /cc @hflynn

hflynn commented:
I think you are going to need to talk me through this @sbesson but it'll need to wait until I finish bean-counting for Jason's reporting.

manics commented on autogen.sh:
```diff
@@ -0,0 +1,33 @@
+sed 1,3d advanced-help.txt > omero/downloads/inplace/advanced-help.txt
+$WORKSPACE/OMERO.clients/importer-cli -h 2> omero/downloads/cli/help.out || echo "Dumped"
+
+echo "Generating Web configuration temapltes"
```
Typo: temapltes

rleigh-dundee commented on autogen.sh:
```diff
@@ -0,0 +1,33 @@
+set -u
+set -e
+set -x
+WORKSPACE=${WORKSPACE:-$pwd}
```
What does this do?  $pwd is undefined.  Should it not be `$(pwd)` ?

rleigh-dundee commented on autogen.sh:
```diff
@@ -0,0 +1,33 @@
+
+echo "Generating configuration properties page"
+$WORKSPACE/OMERO.server/bin/omero config parse --rst | sed "s|$WORKSPACE|/home/omero|" > omero/sysadmins/config.txt
+mkdir -p omero/downloads/ldap
```
Newline here; the mkdir is associated with the next block.

rleigh-dundee commented:
Overall looks OK.  Some additional comments:

- The name.  `autogen.sh` is very widely used for other purposes (google it!), so a more descriptive name would be good.
- Is this the best repository for it to be in?  Would it be more appropriate to locate it in openmicroscopy.git given that it uses content from there?

rleigh-dundee commented:
Looks fine to me and good to merge.  The only minor point would be the unnecessary use of a `.sh` extension.

sbesson commented:
@rleigh-dundee: pushed some fixes, renamed and moved the script under `omero/`. Also added some basic information to the Readme extension /cc @hflynn .

Re the best repository, putting the script under `openmicroscopy.git` would mean more fiddling with submodules to modify the auto-generated content but also at the CI level cloning the upstream repository in the auto-generation job as opposed to retrieving the deliverables as of now.

rleigh-dundee commented:
@sbesson All makes sense and all the latest changes look good.  Good to merge from me.

hflynn commented:
Read me section is fine, happy to merge once everything is green.

joshmoore commented:
I'm still wondering when/how this will be automatedly tested.

sbesson commented:
@joshmoore: discussed with @hflynn earlier. My thoughts in terms of CI:
- http://ci.openmicroscopy.org/job/OMERO-5.1-merge-docs-autogen/ will use the merge artifacts and the merge version of this script (i.e. testing PRs modifying it)
- http://ci.openmicroscopy.org/job/OMERO-5.1-latest-docs-autogen/ will use the latest artifacts and the latest version of this script
I will amend the CI documentation once this is merged.

hflynn commented:
@joshmoore does this mean you want me to hold off merging?

joshmoore commented:
@hflynn : no, just that I'd be for it being tested regularly which is what https://github.com/openmicroscopy/ome-documentation/pull/1083#issuecomment-72005328 is about. So if you're happy with that plan, gfi.

hflynn commented:
I'm assuming you are going to rebase this?

sbesson commented:
Happy to do so although the auto-generated components are different between `dev_5_0` and `develop` so I'll need to remove elements

sbesson commented:
See https://ci.openmicroscopy.org/job/OMERO-5.1-latest-docs-autogen/349/console for the application of this script.

hflynn commented:
If rebasing is going to be more trouble than it is worth, we could leave it as develop only. I guess this is mostly to make adding new autogen stuff easier and that shouldn't really be happening on dev_5_0 any more anyway I suppose, or at least not once 5.1.0 is out.

sbesson commented:
--rebased-to #1084
